File: kitty-items-go/services/kibbles.go
package services

import (
	"context"
	"fmt"
	"log"

	"github.com/dapperlabs/kitty-items-go/templates"
	"github.com/onflow/cadence"
	"github.com/onflow/flow-go-sdk"
)

type KibblesService struct {
	flowService *FlowService
}

func NewKibbles(service *FlowService) *KibblesService {
	return &KibblesService{service}
}

// Mint sends a transaction to the Flow blockchain and returns the generated transactionID as a string.
func (k *KibblesService) Mint(ctx context.Context, destinationAddress flow.Address, amount uint) (string, error) {
	log.Printf("minting kibbles to address=%s", destinationAddress.String())
	sequenceNumber, err := k.flowService.GetMinterAddressSequenceNumber(ctx)
	if err != nil {
		return "", fmt.Errorf("error getting sequence number = %w", err)
	}

	referenceBlock, err := k.flowService.client.GetLatestBlock(ctx, true)
	if err != nil {
		return "", fmt.Errorf("error getting reference block = %w", err)
	}

	tx := flow.NewTransaction().
		SetScript([]byte(templates.MintKibblesTemplate)).
		SetProposalKey(k.flowService.minterAddress, k.flowService.minterAccountKey.Index, sequenceNumber).
		SetPayer(k.flowService.minterAddress).
		SetReferenceBlockID(referenceBlock.ID).
		SetGasLimit(100)

	if err := tx.AddArgument(cadence.NewAddress(destinationAddress)); err != nil {
		return "", err
	}

	if err := tx.AddArgument(cadence.NewUInt(amount)); err != nil {
		return "", err
	}

	return k.flowService.Send(ctx, tx)
}

File: kitty-items-node/src/services/kibbles.ts
import * as fcl from "@onflow/fcl";
import * as t from '@onflow/types';
import * as flowService from './flow';
import * as templates from '../templates/templates';

export const mint = async ({ receiverAddr, amount }) => {
  try {
    const authorization = flowService.authorize(process.env.MINTER_FLOW_ADDRESS, process.env.MINTER_ACCOUNT_KEY_IDX, process.env.MINTER_PRIVATE_KEY)
    const response = await fcl.send([
      fcl.transaction`
        ${templates.mintKibblesTemplate}
      `,
      fcl.args([
        fcl.arg(receiverAddr, t.Address),
        fcl.arg(amount, t.UInt),
      ]),
      fcl.proposer(authorization),
      fcl.payer(authorization),
      fcl.limit(100)
    ]);

    return await fcl.tx(response).onceExecuted();
  } catch (e) {
    console.log(e);
    throw e;
  }
}
